### DISCUSSION TITLE
Is there a way to access ReqData<T> from middleware?

### DISCUSSION THREAD

**Discussion by tyrsday29** [actix/actix-web#3437] [Q&A] [answered]
Hey hope someone can help me out here!

I was trying to access some data I inserted with middleware using 

`    let a = req.extensions().get::<T>().map(|T| T.to_owned());
`

I know that the data has been inserted because I access it in my handlers, but in the middleware it always comes up as none, hopefully I'm just doing something wrong, or is this not possible?

**Comment by robjtede** [comment, marked as answer]
Possible it's being set at an inner layer to the middleware being run. All handlers are run after middleware, but [middleware registration order matters](https://docs.rs/actix-web/latest/actix_web/middleware/#ordering) in this way. For example:

```rust
.wrap(A) // <- sets req data
.wrap(B) // <- outer layer ("wraps" things before it); can't access req data
```

```rust
.wrap(B) // <- can access req data
.wrap(A) // <- sets req data
```

**Comment by tyrsday29** [comment]
That is 100% the issue, I didn't realize it works from the inside out, but that makes sense. Thanks so much!!
